packages/devtools_app/lib/src/performance/performance_controller.dart

```diff
@@ -208,6 +295,16 @@ class PerformanceController
       return;
     }
 
+    if (!frame.isWellFormed &&
+        (_firstWellFormedFrameMicros == null ||
+            frame.timeFromFrameTiming.start.inMicroseconds >=
+                _firstWellFormedFrameMicros)) {
+      // Only try to pull timeline events for frames that are after the first
+      // well formed frame. Timeline events that occurred before this frame will
+      // have already fallen out of the buffer.
+      await processAvailableEvents();
```

Comment:
Are we concerned this will hide timeline events on startup that occur before the first flutter frame (assuming such events can exist)?
We could always add an artificial "before startup" frame to disambiguate this case from the case you are probably trying to avoid of incomplete frames due to full buffers.

Reply:
The timeline events that occur before the first frame will still be present in the timeline flame chart. This check is to prevent us from trying to process new events for frames that we know occurred before the first frame-related timeline events that we have data for. This code is only triggered when a frame is selected, so it is not responsible for the initial load of timeline events.